### PR TITLE
Make the # of backups at each level configurable

### PR DESCRIPTION
**Pull request by assistcontrol**
I'm curious whether anybody else is into this idea. The defaults are so completely sensible that I'm not sure anybody else would be keen to change them.

A setting of "0" means to keep indefinitely.

With it now being done three times (adding in yearly), the
deletion loop becomes repetitive. This commit factors it
out into a function.

### CODE REVIEW

**Comment by alexjurkiewicz** [comment]
Using a common function for the deletion loop seems like a good idea even if the parameters aren't exposed.

File: acts [hunk 1]
```diff
@@ -146,10 +153,10 @@ backuprc=0 # Notice any failed backups
 for dir in $backuptargets; do
     log_debug "message=\"Starting backup of $dir\""
     archive_starttime=$(date +%s)
-    nicedirname=$(echo "$dir" | tr -d '/')
+    nicedirname=${dir:+-${dir#/}}  # /foo -> -foo, / -> ''
```
Comment (alexjurkiewicz) [inline acts]: `$dir` could have multiple `/` in it, for example a backuptarget of `home/aj`, and the new version doesn't seem to handle that:

```
$ docker run -ti alpine sh
/ # dir="home/aj" ; echo ${dir:+-${dir#/}}
-home/aj
```

File: acts [hunk 2]
```diff
@@ -180,40 +187,35 @@ if [ "$backuprc" != "0" ]; then
...
-                log_debug "message=\"Deleting backup $archivetodel\""
-                $tarsnap -d -f "$archivetodel"
-            done
-        done
-    else
-        log_debug "message=\"Found $(echo "$dailybackups" | wc -l) daily backups, not deleting any\""
-    fi
+for dir in $backuptargets; do
+    log_debug "message=\"Checking $dir for old backups to delete\""
+    nicedirname=${dir:+-${dir#/}}  # /foo -> -foo, / -> ''
```
Comment (alexjurkiewicz) [inline acts]: is this local variable no longer used?

File: acts (hunk 1 above)
Comment (assistcontrol) [reply acts]: Ah! Yes, I misread it. I figured only the first / needed to be deleted.

**Comment by assistcontrol** [comment]
I'm gradually working on a copy model that will obviate the need for the loop. It'll do a daily backup every time, and then use tarsnap's copy function to create monthly/yearly backups as needed.

File: acts (hunk 2 above)
Comment (assistcontrol) [reply acts]: Are you referring to `$nicedirname`? The `prune_backups` function picks it up.

**Comment by alexjurkiewicz** [comment]
I saw that comment, sounds clever. I've just pushed a "fix" for Travis too,
which should help.

File: acts (hunk 2 above)
Comment (alexjurkiewicz) [reply acts]: yeah. I misread and thought it was being set twice, I think

**Comment by assistcontrol** [comment]
> I'm gradually working on a copy model that will obviate the need for the loop. It'll do a daily backup every time, and then use tarsnap's copy function to create monthly/yearly backups as needed.

That's a lie. The copy model only affects backup, not pruning. This PR is completely separate.

**Pull request closed without merging**